thewalrus/tests/conftest.py

```diff
@@ -56,6 +56,6 @@ def _wrapper(n, fill_factor=1.0):
         if not np.issubdtype(dtype, np.complexfloating):
             A = A.real
 
-        return dtype(A)
+        return A.astype(dtype)
```

Comment:
Little changes like this, the use of `.item()`, and the use of `np.max`, are to reduce the number of warnings that show up in tests.